### PR TITLE
build: build less/css faster & without grunt

### PR DESCRIPTION
**Pull request by robertkowalski**
this slices our first build step out of the monolithic grunt build.

improvements:

 - faster less build
 - no useless css files for modules in debug folder, just one
   `index.css`
 - multiple targets possible (so we don't need to build from
   `dist/debug` in the future

performance for less build & concat:

```
old:
 - 762ms -- 757ms + 5ms (separate concat)

new:
 - 597ms
```

why is it faster?

1. previous buildstep wrote css files on the disk to read & concat
it later

2. grunt.readfile / grunt.readjson are slow compared to the native
require function which also supports caching

needs node 4.x like our travis builds

### CODE REVIEW

File: build-helper/less.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+'use strict';
```
Comment (benkeen) [inline build-helper/less.js]: This necessary? Just curious - I thought all parsing of the JS files was jshinted - or is that only the application code?

**Comment by benkeen** [comment]
This looks great, Robert. Works fine. Nice to see some ES6 code in Fauxton. :+1: +1

File: build-helper/less.js (hunk 1 above)
Comment (robertkowalski) [inline build-helper/less.js]: yes, some ES6 features like const just work in strict mode right now

**Comment by robertkowalski** [comment]
merged as a0c0b166d85d5e3db30342949290b33107e9304e - yay!

**Pull request closed without merging**